app/assets/javascripts/shopify_app/redirect.js

```diff
@@ -12,15 +15,23 @@
       // If the current window is the 'parent', change the URL by setting location.href
       window.top.location.href = targetInfo.url;
     } else {
-      // If the current window is the 'child', change the parent's URL with postMessage
+      // If the current window is the 'child' or embedded, change the parent's URL with App Bridge redirect
+      // This case happens when the app is installed but the app does not know that because there's no shop entry
       normalizedLink = document.createElement('a');
       normalizedLink.href = targetInfo.url;
 
-      data = JSON.stringify({
-        message: 'Shopify.API.remoteRedirect',
-        data: {location: normalizedLink.href}
+      var AppBridge = window['app-bridge'];
+      var actions = window['app-bridge'].actions;
+      var apiKey = bodyElement.dataset.apiKey;
+      var urlParams = new URLSearchParams(window.location.search);
+
+      var app = AppBridge.createApp({
```

Comment:
Should we also pass `forceRedirect: false` here?

Reply:
Good call!